static/js/blueslip_stacktrace.ts

```diff
@@ -0,0 +1,113 @@
+import * as $ from 'jquery';
+import * as _ from 'underscore';
+import * as StackTraceGPS from 'stacktrace-gps';
+import * as ErrorStackParser from 'error-stack-parser';
+
+type StackFrame = ErrorStackParser.StackFrame;
+
+type CleanStackFrame = {
+    full_path: string;
+    show_path: string;
+    function_name: string;
+    line_number: number;
+};
+
+export function clean_stackframe_data(data: StackFrame): CleanStackFrame {
+    // If the file is local, just show the filename.
+    // Otherwise, show the full path starting from node_modules.
+    const idx = data.fileName.indexOf('node_modules');
+    let full_path;
+    let show_path;
+
+    if (idx !== -1) {
+        full_path = show_path = data.fileName.slice(idx);
+    } else {
+        full_path = data.fileName.split('/').slice(-3).join('/');
+        show_path = full_path.slice(full_path.lastIndexOf('/') + 1);
+    }
+
+    // Don't show the full scope in function name
+    let full_fn_name;
+    try {
+        full_fn_name = /(?:[a-z_]+\.[jt]s(?:\?\w+)?\.)(.+)/.exec(data.functionName)[1];
+    } catch {
+        full_fn_name = data.functionName;
+    }
+    const fn_name = full_fn_name.split('.');
+
+    fn_name[fn_name.length - 1] = '<span class="function-name">'
+        + fn_name[fn_name.length - 1].replace('<', '&lt;').replace('>', '&gt;')
```

Comment:
This seems like a weird thing to do: we’re HTML-escaping only the last element of the array, and only partially at that (no `&` ↦ `&amp;`)?

Reply:
This is only for the case `abc.<anonymous>`, otherwise the names are always just alphanumeric and underscores.

Reply:
I’d be more comfortable with an implementation that’s guaranteed to escape everything than an implementation that hasn’t yet been found to forget to escape anything important.

It would be even better to let the template take care of escaping: split at the last `.` and pass the two parts to the template separately.

Reply:
Done